### PR TITLE
Fix all the yellow, orange and red errors in integration tests

### PR DESCRIPTION
**Pull request by nickvergessen**
cc @rullzer @LukasReschke

### CODE REVIEW

File: build/integration/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -165,11 +184,14 @@ public function asGetsPropertiesOfFolderWith($user, $path, $propertiesTable) {
...
 	 * @Then the single response should contain a property :key with value :value
+	 * @param string $key
+	 * @param string $expectedValue
+	 * @throws \Exception
 	 */
 	public function theSingleResponseShouldContainAPropertyWithValue($key, $expectedValue) {
 		$keys = $this->response;
 		if (!isset($keys[$key])) {
-			throw new \Exception("Cannot find property \"$key\" with \"$expectedalue\"");
+			throw new \Exception("Cannot find property \"$key\" with \"$expectedValue\"");
```
Comment (nickvergessen) [inline build/integration/features/bootstrap/WebDav.php]: :hust:

File: build/integration/features/bootstrap/Sharing.php [hunk 2]
```diff
@@ -439,7 +455,7 @@ public function saveLastShareId()
 	public function shareIdsShouldMatch()
 	{
 		if ($this->savedShareId !== $this->lastShareData['data']['id']) {
-			throw new \Excetion('Expected the same link share to be returned');
+			throw new \Exception('Expected the same link share to be returned');
```
Comment (nickvergessen) [inline build/integration/features/bootstrap/Sharing.php]: :hust:

**Comment by rullzer** [comment]
very nice! :+1:

File: build/integration/federation_features/federated.feature [hunk 3]
```diff
@@ -7,7 +7,7 @@ Feature: federated
 		And user "user1" exists
 		And Using server "LOCAL"
 		And user "user0" exists
-		When User "user0" from server "LOCAL" shares "/textfile0.txt" with user "user1" from server "REMOTE"
+		When User "user0" from local server shares "/textfile0.txt" with user "user1" from remote server
```
Comment (SergioBertolinSG) [inline build/integration/federation_features/federated.feature]: Why this change? All new added tests here are using this format.

Comment (nickvergessen) [inline build/integration/federation_features/federated.feature]: Well the variables "LOCAL" and "REMOTE" are not used. so they make no sense and are marked as unused variable.

Either you specify valid urls, that can be used as server part on the share, or it doesn't need to be a variable.

File: build/integration/features/bootstrap/FederationContext.php [hunk 4]
```diff
@@ -12,14 +10,15 @@
...
 	/**
-	 * @When /^User "([^"]*)" from server "([^"]*)" shares "([^"]*)" with user "([^"]*)" from server "([^"]*)"$/
+	 * @When /^User "([^"]*)" from local server shares "([^"]*)" with user "([^"]*)" from remote server$/
+	 * @param string $userLocal
+	 * @param string $pathLocal
+	 * @param string $userRemote
 	 */
-	public function federateSharing($userLocal, $serverLocal, $pathLocal, $userRemote, $serverRemote){
+	public function federateSharing($userLocal, $pathLocal, $userRemote){
 		$shareWith = "$userRemote@" . substr($this->remoteBaseUrl, 0, -4);
```
Comment (SergioBertolinSG) [inline build/integration/features/bootstrap/FederationContext.php]: It won't work, because there is a bug here, and $serverRemote is used to fix it. shareWith has to use remoteBaseUrl or localBaseUrl, so that variable is needed.

Comment (rullzer) [inline build/integration/features/bootstrap/FederationContext.php]: I don't follow your reasoning. If they are not used then they don't do anything....

Comment (SergioBertolinSG) [inline build/integration/features/bootstrap/FederationContext.php]: Look here https://github.com/owncloud/core/pull/22750/files#diff-7ba81fbf77e91118519e1c537714a487L22

Comment (rullzer) [inline build/integration/features/bootstrap/FederationContext.php]: Aaaah conflicting PR's.
@SergioBertolinSG maybe it would be better to have 2 simple but explitict functions.

One for. 

`@When /^User "([^"]*)" from local server shares "([^"]*)" with user "([^"]*)" from remote server$/`

And one for

`@When /^User "([^"]*)" from remote server shares "([^"]*)" with user "([^"]*)" from local server$/`

Comment (nickvergessen) [inline build/integration/features/bootstrap/FederationContext.php]: okay, I will drop this then, but really should be fixed in the other PR then.

**Comment by MorrisJobke** [comment]
:+1:
